### PR TITLE
Added client_secret.bcheck

### PR DESCRIPTION
**Pull request by LabMC**
### BCheck Contributions

- The "client_secret.bcheck" file has been created to detect for the "client_secret", "client_id", & "refresh_token" OAuth API variables, plus variants of "api-key", within front-end files. This file is placed in the other/API folder.
- Minor update to Content-Security-Policy.bcheck to look for findings in .js & .css files.

* [X] BCheck compiles and executes as expected
* [X] BCheck contains appropriate metadata (name, version, author, description and appropriate tags)
* [X] Only .bcheck files have been added or modified
* [X] BCheck is in the appropriate folder
* [X] PR contains single or limited number of BChecks (Multiple PRs are preferred)
* [X] BCheck attempts to minimize false positives

### CODE REVIEW

File: other/APIs/client_secret.bcheck [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    name: "Use of Known API Variable"
+    description: "This BCheck looks for known API variables (client_secret, client_id, refresh_token, & APIKey)."
+    author: "Kyle Gilligan"
+    tags: "passive", "API", "client_secret", "client_id", "ApiKey"
+
+define:
+    # Issue details (for discovery of insecure API variables) as individual string texts.
+    issueDetail1 = `A known sensitive API parameter has been discovered within a front-end file of this web`
+    issueDetail1a = `application.`
+    issueDetail1FULL = `{issueDetail1} {issueDetail1a}`
```
Comment (PortSwiggerWiener) [inline other/APIs/client_secret.bcheck:12]: These are probably simple enough to merge into a single variable (no need for interpolated strings then as well).

File: other/APIs/client_secret.bcheck [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    iD_clientSecret1 = `\n    • client_secret: This OAuth variable is used to authenticate applications towards`
+    iD_clientSecret2 = `an intended authorization server. Essentially the 'password' of API authorization.`
+    iD_clientSecret3 = `If a user finds both an API's client_id & client_secret, they may be able to access an`
+    iD_clientSecret4 = `endpoint's resources by impersonating the application.`
+    iD_clientSecretFULL = `{iD_clientSecret1} {iD_clientSecret2} {iD_clientSecret3} {iD_clientSecret4}`
+    iD_refreshToken1 = `\n    • refresh_token: Secret variables used by an application to request new access`
+    iD_refreshToken2 = `tokens for an API.`
+    iD_refreshTokenFULL = `{iD_refreshToken1} {iD_refreshToken2}`
```
Comment (PortSwiggerWiener) [inline other/APIs/client_secret.bcheck:20]: These (ID_refreshToken...) are probably simple enough to merge into a single variable (no need for interpolated strings then as well).

File: other/APIs/client_secret.bcheck [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    iD_apiKey1 = `\n    • api_key: An API Key acts as a secret identifier token used to provide identification`
+    iD_apiKey2 = `for an application when using a target API.`
+    iD_apiKeyFULL = `{iD_apiKey1} {iD_apiKey2}`
+    issueRemediation1 = `Unless impossible to remove based on application requirements, these API parameters`
+    issueRemediation1a = `should never list their values within front-end files.`
+    issueRemediation1FULL = `{issueRemediation1} {issueRemediation1a}`
```
Comment (PortSwiggerWiener) [inline other/APIs/client_secret.bcheck:26]: These (issueRemediation...) are probably simple enough to merge into a single variable (no need for interpolated strings then as well).

File: other/APIs/client_secret.bcheck [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    # Issue details (for discovery of known non-sensitive API variables) as individual string texts.
+    issueDetail2FULL = `A known API parameter has been discovered within a front-end file of this web application.`
+    iD_clientId1 = `\n    • client_id: This OAuth variable is used to authenticate applications towards an`
+    iD_clientId2 = `intended authorization server. Essentially the 'username' of API authorization. If a user`
+    iD_clientId3 = `finds both an API's client_id & client_secret, they may be able to access an endpoint's`
+    iD_clientId4 = `resources by impersonating the application.`
+    iD_clientIdFULL = `{iD_clientId1} {iD_clientId2} {iD_clientId3} {iD_clientId4}`
+    issueRemediation2 = `Unless impossible to remove based on application requirements, it becomes recommended`
+    issueRemediation2a = `for this API parameter's value to not be listed in a front-end file.`
```
Comment (PortSwiggerWiener) [inline other/APIs/client_secret.bcheck:35]: These (issueRemediation2...) are probably simple enough to merge into a single variable (no need for interpolated strings then as well).

File: other/APIs/client_secret.bcheck [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    issueRemediation2FULL = `{issueRemediation2} {issueRemediation2a}`
+
+given response then
+    # Nesting several if statements becomes necessary to quickly reduce checks for FPs.
+
+    # This check ensures that only notable 200s HTTP responses appear present in the HTTP response.
+    if ({latest.response.status_code} matches "(200|204|206)") then
+
+        # This check ensures that unacceptable MIME types get ignored to reduce false positives.
+        if not({latest.response.headers} matches "(Content-Type: image/)") then
```
Comment (PortSwiggerWiener) [inline other/APIs/client_secret.bcheck:45]: These first two if blocks could be merged.

**Review by PortSwiggerWiener — COMMENTED** [review]
Many thanks for the submission! Looks really good. Minor comments above.

File: other/APIs/client_secret.bcheck (hunk 2 above)
Comment (LabMC) [reply other/APIs/client_secret.bcheck]: Made changes based on recommendation.

File: other/APIs/client_secret.bcheck (hunk 1 above)
Comment (LabMC) [reply other/APIs/client_secret.bcheck]: Made changes based on recommendation.

File: other/APIs/client_secret.bcheck (hunk 3 above)
Comment (LabMC) [reply other/APIs/client_secret.bcheck]: Made changes based on recommendation.

File: other/APIs/client_secret.bcheck (hunk 4 above)
Comment (LabMC) [reply other/APIs/client_secret.bcheck]: Made changes based on recommendation.

File: other/APIs/client_secret.bcheck (hunk 1 above)
Comment (LabMC) [reply other/APIs/client_secret.bcheck]: Made changes based on recommendation.

File: other/APIs/client_secret.bcheck (hunk 5 above)
Comment (LabMC) [reply other/APIs/client_secret.bcheck]: Made changes based on recommendation.

**Review by PortSwiggerWiener — APPROVED** [review]
Many thanks for the revisions.

Looks good 👍

**Review by josh-psw — APPROVED** [review]

**Review by Hannah-PortSwigger — APPROVED** [review]
Looks good 👍

**Pull request merged**